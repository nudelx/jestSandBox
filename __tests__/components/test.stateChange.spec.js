import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import enzymeSerializer from 'enzyme-to-json/serializer'
expect.addSnapshotSerializer(enzymeSerializer)
Enzyme.configure({ adapter: new Adapter() })

class TestMe extends React.Component {
  state = { show: false }

  onClick = () => this.setState(({ show }) => ({ show: !show }))

  render() {
    return (
      <div>
        <div className="force">
          {this.state.show ? (
            <div className="data">here we go !</div>
          ) : (
            <div className="loading">wait ....</div>
          )}
        </div>
        <button onClick={this.onClick}>clickMe</button>
      </div>
    )
  }
}

describe('test react state change ', () => {
  it('should render the loading and react on button', () => {
    const component = shallow(<TestMe />)
    expect(component).toMatchSnapshot()
    expect(component.state()).toMatchSnapshot()
    component.find('button').simulate('click')
    expect(component).toMatchSnapshot()
    expect(component.state()).toMatchSnapshot()
  })
})
