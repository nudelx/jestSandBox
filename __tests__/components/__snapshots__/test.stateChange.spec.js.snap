// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test react state change  should render the loading and react on button 1`] = `
<div>
  <div
    className="force"
  >
    <div
      className="loading"
    >
      wait ....
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    clickMe
  </button>
</div>
`;

exports[`test react state change  should render the loading and react on button 2`] = `
Object {
  "show": false,
}
`;

exports[`test react state change  should render the loading and react on button 3`] = `
<div>
  <div
    className="force"
  >
    <div
      className="data"
    >
      here we go !
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    clickMe
  </button>
</div>
`;

exports[`test react state change  should render the loading and react on button 4`] = `
Object {
  "show": true,
}
`;
